macro/private/control/yalang-yl620a.g

```diff
@@ -0,0 +1,229 @@
+    set global.arborState[param.S][3] = { var.spindleLimits }
+
+var shouldRun = { (spindles[param.S].state == "forward" || spindles[param.S].state == "reverse") && spindles[param.S].active > 0 }
+
+; 0 = Error
+; 1 = Inverter State
+; 2 = Target Frequency
+; 3 = Current Frequency
+; 4 = Current Current
+; 5 = Current Voltage
+; 6 = Bus Voltage
+; 7 = Number of fields in Multi Rate
+; 8 = Acceleration/Deceleration Flags
+M261.1 P{param.C} A{param.A} F3 R{var.stateBytesAddr} B9 V"stateBytes"
+G4 P{var.cmdWait}
+var stateBytes = { global.returnVal }
+
+if { var.stateBytes == null }
+    echo { "ArborCtl: VFD stopped responding to inputs."}
+    set global.arborState[param.S][4] = true
+    M99
+
+var error = var.stateBytes[0]
+
+if { var.error == null }
+    echo { "ArborCtl: VFD in error state."}
+    set global.arborState[param.S][4] = true
+    M99
+
+var spindlePower = var.stateBytes[4] * var.stateBytes[5]
+
+; Check for VFD errors
+if { var.stateBytes[0] != 0 }
+    echo { "ArborCtl: VFD Error detected. Code=" ^ var.stateBytes[0] }
+    var i = 0
+    while {var.stateBytes[0] > 0 }
+        var bitPresent = { mod(var.stateBytes[0], 2) == 1 }
+        if { var.bitPresent }
+            if {var.i == 0}
+                echo { "ArborCtl: Too Low Voltage - Supply voltage has dropped too low."}
+            elif {var.i == 1}
+                echo { "ArborCtl: Too High Voltage - Supply voltage has risen too high."}
+            elif {var.i == 2}
+                echo { "ArborCtl: Overcurrent - When motor was running, current levels rose too high."}
+            elif {var.i == 3}
+                echo { "ArborCtl: External PWM Error - An error in the CPU's PWM circuitry has occurred."}
+            elif {var.i == 4}
+                echo { "ArborCtl: Short Circuit Alarm - A short circuit has occurred."}
+            elif {var.i == 5}
+                echo { "ArborCtl: External Fault - An external fault has been indicated to the VFD."}
+            elif {var.i == 6}
+                echo { "ArborCtl: Internal Data Storage Error - An error has occurred in the VFD's internal data storage."}
+            elif {var.i == 7}
+                echo { "ArborCtl: Overheating - The VFD is overheating."}
+            elif {var.i == 8}
+                echo { "ArborCtl: Temperature Detection Error - The VFD encountered errors while trying to ascertain its temperature."}
+            elif {var.i == 10}
+                echo { "ArborCtl: Powering Down - The VFD is in power down mode."}
+            elif {var.i == 11}
+                echo { "ArborCtl: RS485 Communication Interrupted - RS485 communication was interrupted for too long."}
+            elif {var.i == 12}
+                echo { "ArborCtl: Parametric Error."}
+            elif {var.i == 15}
+                echo { "ArborCtl: Motor Overheating - The motor is overheating."}
+            else
+                echo { "ArborCtl: Unknown Error."}
+        set var.i = { var.i+1 }
```

Comment:
Can we just use `iterations` for this? It isn't used in the loop conditional but as far as I understand it, will still be maintained and iterated by RRF.